src/Integration/RequestIntegration.php

```diff
@@ -109,8 +132,64 @@ private function removePiiFromHeaders(array $headers): array
     {
         $keysToRemove = ['authorization', 'cookie', 'set-cookie', 'remote_addr'];
 
-        return array_filter($headers, function ($key) use ($keysToRemove) {
-            return !\in_array(strtolower($key), $keysToRemove, true);
-        }, ARRAY_FILTER_USE_KEY);
+        return array_filter(
+            $headers,
+            static function (string $key) use ($keysToRemove): bool {
+                return !\in_array(strtolower($key), $keysToRemove, true);
+            },
+            ARRAY_FILTER_USE_KEY
+        );
+    }
+
+    /**
+     * Gets the decoded body of the request, if available. If the Content-Type
+     * header contains "application/json" then the content is decoded and if
+     * the parsing fails then the raw data is returned. If there are submitted
+     * fields or files, all of their information are parsed and returned.
+     *
+     * @param ServerRequestInterface $serverRequest The server request
+     *
+     * @return mixed
+     */
+    private function captureRequestBody(ServerRequestInterface $serverRequest)
+    {
+        $maxRequestBodySize = $this->options->getRequestBodies();
+        $requestBody = $serverRequest->getBody();
+
+        if ('none' === $maxRequestBodySize || ('small' === $maxRequestBodySize && $requestBody->getSize() > self::REQUEST_BODY_SMALL_MAX_CONTENT_LENGTH) || ('medium' === $maxRequestBodySize && $requestBody->getSize() > self::REQUEST_BODY_MEDIUM_MAX_CONTENT_LENGTH)) {
```

Comment:
Please split this on multiple lines for better readability:

```suggestion
        if (
        'none' === $maxRequestBodySize 
        || ('small' === $maxRequestBodySize && $requestBody->getSize() > self::REQUEST_BODY_SMALL_MAX_CONTENT_LENGTH) 
        || ('medium' === $maxRequestBodySize && $requestBody->getSize() > self::REQUEST_BODY_MEDIUM_MAX_CONTENT_LENGTH)
        ) {
```

Reply:
Or even better, we could split this in a private method with a switch of the enum.